python/podio/podio_config_reader.py

```diff
@@ -85,18 +85,25 @@ def _bare_member_conv(result):
 
   def parse(self, string, require_description=True):
     """Parse the passed string"""
-    matchers_cbs = [
+    default_matchers_cbs = [
         (self.full_array_re, self._full_array_conv),
-        (self.member_re, self._full_member_conv)
-        ]
-
-    if not require_description:
-      matchers_cbs.extend((
-          (self.bare_array_re, self._bare_array_conv),
-          (self.bare_member_re, self._bare_member_conv)
-          ))
-
-    return self._parse_with_regexps(string, matchers_cbs)
+        (self.member_re, self._full_member_conv)]
+
+    no_desc_matchers_cbs = [
+        (self.bare_array_re, self._bare_array_conv),
+        (self.bare_member_re, self._bare_member_conv)]
+
+    if require_description:
+      try:
+        return self._parse_with_regexps(string, default_matchers_cbs)
+      except DefinitionError:
+        """check whether we could parse this if we don't require a description and
+        provide more details in the error if we can"""
+        self._parse_with_regexps(string, no_desc_matchers_cbs)
```

Comment:
```suggestion
        self._parse_with_regexps(string, no_desc_matchers_cbs)
        # pylint: disable-next=raise-missing-from
```
To silence the pylint warning which is more of a false positive in this case